binaries/runtime/src/operator/shared_lib.rs

```diff
@@ -27,26 +27,21 @@ use tokio::sync::{mpsc::Sender, oneshot};
 use tracing::{field, span};
 
 pub fn run(
-    node_id: &NodeId,
-    operator_id: &OperatorId,
+    _node_id: &NodeId,
+    _operator_id: &OperatorId,
     source: &str,
     events_tx: Sender<OperatorEvent>,
     incoming_events: flume::Receiver<Event>,
     init_done: oneshot::Sender<Result<()>>,
 ) -> eyre::Result<()> {
     let path = if source_is_url(source) {
-        let target_path = adjust_shared_library_path(
-            &Path::new("build")
-                .join(node_id.to_string())
-                .join(operator_id.to_string()),
-        )?;
+        let target_path = &Path::new("build");
```

Comment:
The reason for this `adjust_shared_library_path` was that you can provide cross-platform URLs. E.g. `example.com/operator` would become `example.com/liboperator.so` on Linux and `example.com/operator.dll` on Windows. This is a bit hacky, but without this adjust you need different dataflow files for Windows and Linux.

Reply:
I understand, but I honestly find it super hard to keep track about all the rules that we had rather than having simple link. 

I would rather prefer user implement their own url resolver for cross platform than choosing a url definition that people might need to remember.

this might resolve some issues as you might also want to deal with architecture (x86 or arm) and I don't think we should expect people to build library for all platform.

Reply:
Let say I want to use this feature for something like: 

-> `libopencv_plot.so`
-> I need to push into a server 
-> I need to remember to put in my node: `url.com/opencv_plot` which is not obvious
-> And if there is a bug from me or someone else it's going to be: Something went wrong with `build/libplot.so` for example. Which I need to remember links to `url.com/opencv_plot` which is my `libopencv_plot.so`

Knowing that none of this is going to be properly documented in the near future...

Reply:
Being cross platform could be as simple as:

```
  - id: plot
    build: }
        wget -O plot.zip example.com/plot/$OSTYPE/node/plot.zip
        unzip plot.zip
    path: libplot.so
    inputs:
      image: camera/image
```

Reply:
I fully agree that the status quo solution was hacky and not ideal. Of course, things get simpler if we simply ignore the platform differences. However, if we want dora to be cross-platform, we should provide _some_ way to use this feature for cross-platform dataflows. 

> Being cross platform could be as simple as:
>
> ```yaml
> [...]
> path: libplot.so
> [...]
> ```

The issue is that the file would need to be named `plot.dll` if you want to run it on Windows.

Reply:
So in windows you can run an exe without the `.exe` suffix so we can always call something like `target/release/node` without the exe extension. So I would prefer defaulting to this.

See: https://stackoverflow.com/questions/44441265/calling-an-exe-without-the-extension-but-with-i

FYI, i hope that most binary that we ship for dora can be packaged in a way that is `pip` or `cargo` installable so that everything is contained in PATH by default and easy to find and deploy.

Reply:
That's useful for local paths, but unfortunately not for URLs. We need different binaries on Windows and Linux, so downloading the same file would not help. Also, the downloaded file would probably need an `.exe` extension to be recognized as an executable file on Windows, no? 

However, even if we added a `.exe` extension automatically, we would still have the issue that there are different CPU architectures etc. So placeholders is probably the way to go, e.g. `example.com/path/{target}/node`, which would expand to `example.com/path/x86_64-unknown-linux-gnu/node` on an `x86_64` Linux system. Alternatively, we could of course support separate URLs for each target, e.g.:

```yaml
path:
  x86_64-unknown-linux-gnu: https://example.com/path/x86/linux/node
  x86_64-pc-windows-msvc: https://example.com/path/x86/win/node.exe
```

(Of course, this is not a high priority right now.)